com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PrepareAppsRestartStep.java

```diff
@@ -22,6 +24,8 @@ protected StepPhase executeStep(ExecutionWrapper execution) {
             .setVariable(Constants.EXECUTE_ONE_OFF_TASKS, false);
         StepsUtil.setUseIdleUris(execution.getContext(), false);
         StepsUtil.setSkipUpdateConfigurationEntries(execution.getContext(), false);
+        
+        StepsUtil.setCompletedAppsInParallel(execution.getContext(), Collections.emptyList());
```

Comment:
This default should be in the getter IMO. That way people would never forget to set it.

Reply:
I am not sure I understood this statement but the context already includes all applications after the deployment, so this is necessary to be set to empty in order for restart iteration in bg-deploy to reiterate them.

Reply:
Aaaaaahhhh, okaay! Now it makes sense. However, this is also a bit dirty, because this step is basically cleaning up all the shit from the previous steps... Can't this be avoided?

Btw, I meant that if the getter returned a default (Collections.emptyList()) in case the value in the context was null, then you wouldn't need to set it to Collections.emptyList() explicitly here just to avoid a NPE. However, I was under the assumption that this is the first place where we used this variable, which is apparently not true.